### ISSUE TITLE
Upgrade to React Router (MERN 2.6) 

### ISSUE THREAD

**Issue by mannyhenri** [Hashnode/mern-starter#399]
Main thread related to upgrading to MERN 2.6 (react router 4)

**Comment by mannyhenri** [comment]
@ashifa454 here is where we can track ongoing issues with React router upgrade (2.6)

**Comment by ashifa454** [comment]
Ok, i am going to continue from today. Was kinda stuck at companies
deployment.

**Comment by ashifa454** [comment]
can you guide me why we are using `match` in server.js, what is the alternative for it in V4

**Comment by mattlubner** [comment]
@ashifa454 I don't believe there's a direct equivalent for `match` on the server. We'll have to use a `<StaticRouter />` and pass in `context` and `location`, then after calling `renderToString()` we can check for `context.url` and handle redirects that way.

I'm not completely sure about errors, but I'm guessing wrapping the whole thing in a try-catch or Promise should suffice for handling render errors.

You can read a bit more about it here: https://reacttraining.com/react-router/web/guides/server-rendering

**Comment by ashifa454** [comment]
i am following the official docs and have resolved the `match` thing, just trying to fix checksum, for server and client

**Comment by Amdadullah-Raihan** [comment]
Hi there- Was this issue solved? if not I want to contribute.
